### ISSUE TITLE
[Bug] - Random crash while downloading

### ISSUE THREAD

**Issue by sergiohzph** [Crunchy-DL/Crunchy-Downloader#95]
Vídeo showing what happens:
https://github.com/user-attachments/assets/acc539d3-4edf-4741-b220-b2c26770f69e

.NET Crash Log:
```
Log Name:      Application
Source:        Application Error
Date:          22/08/2024 23:55:07
Event ID:      1000
Task Category: Application Crashing Events
Level:         Error
Keywords:      
User:          DESKTOP-DCHT2C2\Sergio
Computer:      DESKTOP-DCHT2C2
Description:
Faulting application name: CRD.exe, version: 1.5.23.0, time stamp: 0x65f90000
Faulting module name: KERNELBASE.dll, version: 10.0.22621.3958, time stamp: 0xfbc3a4f6
Exception code: 0xe0434352
Fault offset: 0x000000000005fabc
Faulting process id: 0x0x5EB0
Faulting application start time: 0x0x1DAF4DDD3FFEFBA
Faulting application path: D:\Software\Crunchyroll-Downloader\CRD.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: f2c2a55f-d605-4883-97aa-9bd4b85cdcb8
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" />
    <EventID>1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2024-08-22T21:55:07.7222791Z" />
    <EventRecordID>4312</EventRecordID>
    <Correlation />
    <Execution ProcessID="21508" ThreadID="1900" />
    <Channel>Application</Channel>
    <Computer>DESKTOP-DCHT2C2</Computer>
    <Security UserID="S-1-5-21-3977021943-1919960476-2993769995-1000" />
  </System>
  <EventData>
    <Data Name="AppName">CRD.exe</Data>
    <Data Name="AppVersion">1.5.23.0</Data>
    <Data Name="AppTimeStamp">65f90000</Data>
    <Data Name="ModuleName">KERNELBASE.dll</Data>
    <Data Name="ModuleVersion">10.0.22621.3958</Data>
    <Data Name="ModuleTimeStamp">fbc3a4f6</Data>
    <Data Name="ExceptionCode">e0434352</Data>
    <Data Name="FaultingOffset">000000000005fabc</Data>
    <Data Name="ProcessId">0x5eb0</Data>
    <Data Name="ProcessCreationTime">0x1daf4ddd3ffefba</Data>
    <Data Name="AppPath">D:\Software\Crunchyroll-Downloader\CRD.exe</Data>
    <Data Name="ModulePath">C:\Windows\System32\KERNELBASE.dll</Data>
    <Data Name="IntegratorReportId">f2c2a55f-d605-4883-97aa-9bd4b85cdcb8</Data>
    <Data Name="PackageFullName">
    </Data>
    <Data Name="PackageRelativeAppId">
    </Data>
  </EventData>
</Event>
```
Crash log file:
[logfile.txt](https://github.com/user-attachments/files/16719382/logfile.txt)

**Comment by Elwador** [comment]
do you have a different .NET runtime error in the event view applications thing - because that doesn't say why it crashed

**Comment by Elwador** [comment]
should be fixed with - [v1.5.24](https://github.com/Crunchy-DL/Crunchy-Downloader/releases/tag/v1.5.24)
